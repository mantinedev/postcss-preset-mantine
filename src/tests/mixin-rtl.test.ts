import { testTransform } from './utils';

const baseInput = `
.demo {
  @mixin rtl {
    margin-right: 1rem;
  }

  @mixin not-rtl {
    margin-left: 1rem;
  }
}
`;

const whereInput = `
.demo {
  @mixin where-rtl {
    margin-right: 1rem;
  }

  @mixin where-not-rtl {
    margin-left: 1rem;
  }
}
`;

describe('mixin-rtl', () => {
  it('transforms rtl mixins correctly', async () => {
    const res = await testTransform(baseInput);
    expect(res.css).toMatchSnapshot();
  });

  it('transforms where-rtl mixins correctly', async () => {
    const res = await testTransform(whereInput);
    expect(res.css).toMatchSnapshot();
  });
});
