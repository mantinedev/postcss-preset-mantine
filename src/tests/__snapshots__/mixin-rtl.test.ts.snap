// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixin-rtl transforms rtl mixins correctly 1`] = `
"
[dir="rtl"] .demo {
    margin-right: 1rem
}
  :root:not([dir="rtl"]) .demo {
    margin-left: 1rem
}
"
`;

exports[`mixin-rtl transforms where-rtl mixins correctly 1`] = `
"
:where([dir="rtl"]) .demo {
    margin-right: 1rem
}
  :where([dir="ltr"]) .demo {
    margin-left: 1rem
}
"
`;
